src/thumbnail/main_generate_thumbnail.py

```diff
@@ -0,0 +1,470 @@
+    draw.text(text_coordinates, current_round,
+              (255, 255, 255), font=font, anchor="mm", stroke_width=round(stroke_width*(actual_text_size/text_size)), stroke_fill=outline_color)
+
+
+def paste_main_icon(thumbnail, icon_path):
+    if icon_path:
+        max_x_size = round(thumbnail.size[0]*(150.0/1920.0))
+        max_y_size = round(thumbnail.size[1]*(150.0/1080.0))
+        max_size = (max_x_size, max_y_size)
+
+        icon_image = Image.open(icon_path).convert('RGBA')
+        icon_size = calculate_new_dimensions(icon_image.size, max_size)
+        icon_image = icon_image.resize(icon_size, resample=Image.BICUBIC)
+
+        icon_x = round(thumbnail.size[0]/2 - icon_size[0]/2)
+        icon_y = round(thumbnail.size[1]*(6.0/1080.0))
+        icon_coordinates = (icon_x, icon_y)
+        composite_image = create_composite_image(
+            icon_image, thumbnail.size, icon_coordinates)
+        thumbnail = Image.alpha_composite(thumbnail, composite_image)
+    return(thumbnail)
+
+def paste_side_icon(thumbnail, icon_path_list):
+    if len(icon_path_list) > 2:
+        raise(ValueError(msg="Error: icon_path_list has 3 or more elements"))
+
+    max_x_size = round(thumbnail.size[0]*(100.0/1920.0))
+    max_y_size = round(thumbnail.size[1]*(100.0/1080.0))
+    max_size = (max_x_size, max_y_size)
+    icon_y = round(thumbnail.size[1]*(10.0/1080.0))
+
+    for index in range(0, len(icon_path_list)):
+        icon_path = icon_path_list[index]
+        if icon_path:
+            icon_image = Image.open(icon_path).convert('RGBA')
+            icon_size = calculate_new_dimensions(icon_image.size, max_size)
+            icon_image = icon_image.resize(icon_size, resample=Image.BICUBIC)
+
+            icon_x = index*round(thumbnail.size[0] - icon_size[0])
+            x_offset = -round(thumbnail.size[0]*(10.0/1920.0)) * ((index*2)-1)
+            icon_x = icon_x + x_offset
+
+            icon_coordinates = (icon_x, icon_y)
+            composite_image = create_composite_image(
+                icon_image, thumbnail.size, icon_coordinates)
+            thumbnail = Image.alpha_composite(thumbnail, composite_image)
+    return(thumbnail)
+
+def generate():
+    foreground_path = "./scripts/thumbnail_base/foreground.png"
+    background_path = "./scripts/thumbnail_base/background.png"
+    data_path = "./out/program_state.json"
+    out_path = "./out/thumbnails"
+    tmp_path = "./tmp"
+    main_icon_path = "./assets/icons/icon.png"
+    side_icon_list = ["", ""]
+
+    font_list = ["./assets/font/OpenSans/OpenSans-Bold.ttf", "./assets/font/OpenSans/OpenSans-Semibold.ttf"]
+
+    with open(data_path, 'rt', encoding='utf-8') as f:
+        data = json.loads(f.read())
+
+    game_codename = data.get("game").get("codename")
+    asset_data_path = f"./user_data/games/{game_codename}/{used_assets}/config.json"
+    with open(asset_data_path, 'rt', encoding='utf-8') as f:
+        all_eyesight = json.loads(f.read()).get("eyesights")
```

Comment:
Proposed fix for eyesight issues: Add on line 423:
```
        global all_eyesight
```

Reply:
Done !